tests/core/nlg/test_response.py

```diff
@@ -80,31 +80,127 @@ async def test_nlg_when_multiple_conditions_satisfied():
     assert resp.get("text") in ["example A", "example B"]
 
 
-@pytest.mark.parametrize(("response_text"), ("example a",))
-async def test_nlg_slot_case_sensitivity(response_text: Text):
-    responses = {
-        "utter_action": [
-            {
-                "text": response_text,
-                "condition": [{"type": "slot", "name": "test", "value": "cold"}],
-            }
-        ]
-    }
-
-    t = TemplatedNaturalLanguageGenerator(responses=responses)
+@pytest.fixture(scope="session")
+def test_slots() -> List[object]:
     slot_a = CategoricalSlot(
         name="test",
         mappings=[{"type": "from_text", "value": ["cold", "hot"]}],
         initial_value="Cold",
         influence_conversation=False,
     )
-    tracker = DialogueStateTracker(sender_id="test_nlg", slots=[slot_a])
+    slot_b = BooleanSlot(
+        name="test2",
+        mappings=[{}],
+        initial_value=False,
+        influence_conversation=False,
+    )
+    slot_c = CategoricalSlot(
+        name="test3",
+        mappings=[{"type": "from_text", "value": ["cold", "hot"]}],
+        initial_value="hot",
+        influence_conversation=False,
+    )
+    return [slot_a, slot_b, slot_c]
+
+
+@pytest.fixture(scope="session")
+def test_responses() -> List[Dict[Text, List[Dict[Text, Any]]]]:
+    return [
+        {
+            "utter_action": [
+                {
+                    "text": "example a",
+                    "condition": [{"type": "slot", "name": "test", "value": "cold"}],
+                }
+            ]
+        },
+        {
+            "utter_action_multiple_conditions": [
+                {
+                    "text": "example a",
```

Comment:
I'd also give this a different text to make it clear that these different responses are different.